src/pages/profile/RemoteProfileView.tsx

```diff
@@ -0,0 +1,184 @@
+import { useEffect, useState } from "react";
+import { ReactElement } from "react";
+import { useParams, Navigate } from "react-router-dom";
+import {
+  useProfileDetailsQuery,
+  useMyDetailsQuery,
+  useGetProfileFollowingQuery,
+  usePostProfileFollowMutation,
+  useDelProfileFollowMutation,
+} from "@/entities/profile/api/profileApi";
+import { Preloader } from "@/shared/ui/Preloader";
+import { ProfileInfo } from "@/widgets/Profile/ProfileInfo";
+import { EventsList } from "@/widgets/EventsList";
+import { Button } from "@/shared/ui/Buttons/Button";
+import isEqual from "lodash/isEqual";
+import NonFound from "@/pages/errors/NonFound";
+import { ProfileFollowButton } from "@/widgets/ProfileButton";
+import {
+  useGetProfileCreatedEventsQuery,
+  useGetProfilePlannedEventsQuery,
+  useGetProfileVisitedEventsQuery,
+} from "@/widgets/Profile/ProfileEvents/api/profileEventsApi";
+
+
+function RemoteProfileView(): ReactElement {
+  const [isPageReady, setIsPageReady] = useState(false);
+  const { userId = "0" } = useParams();
+  const {
+    data: profileData,
+    isLoading: isLoadingRemoteUser,
+    isSuccess: isSuccessRemoteUser,
+    isError: isErrorRemoteUser,
+  } = useProfileDetailsQuery({ userId: userId });
+  const isPrivateUser = profileData?.is_private;
+
+  const {
+    data: createdEvents = [],
+    isLoading: isCreatedEventsLoading,
+    isSuccess: isCreatedEventsSuccess,
+    isError: isCreatedEventsError,
+    error: eventsCreatedError,
+  } = useGetProfileCreatedEventsQuery(
+    { userId: String(profileData?.id) },
+    { skip: !isSuccessRemoteUser }
+  );
+  const {
+    data: plannedEvents = [],
+    isLoading: isPlannedEventsLoading,
+    isSuccess: isPlannedEventsSuccess,
+    isError: isPlannedEventsError,
+    error: eventsPlannedError,
+  } = useGetProfilePlannedEventsQuery(
+    { userId: String(profileData?.id) },
+    { skip: !isSuccessRemoteUser }
+  );
+  const {
+    data: visitedEvents = [],
+    isLoading: isVisitedEventsLoading,
+    isSuccess: isVisitedEventsSuccess,
+    isError: isVisitedEventsError,
+    error: eventsVisitedError,
+  } = useGetProfileVisitedEventsQuery(
+    { userId: String(profileData?.id) },
+    { skip: !isSuccessRemoteUser }
+  );
+
+
+  const { data: currentProfileData, isLoading: isLoadingCurrentUser } = useMyDetailsQuery();
+
+  const {
+    data: profileFollow = [],
+    isLoading: isLoadingProfileFollow,
+    isSuccess: isSuccessFollowing,
+  } = useGetProfileFollowingQuery(
+    {
+      userId: String(currentProfileData?.id),
+    },
+    {
+      skip: !currentProfileData,
+    }
+  );
+
+  const [userProfileFollowStatus, setUserProfileFollowStatus] = useState<string | undefined>(undefined);
+
+  const getProfileFollowStatus = (id: string) =>
```

Comment:
userId is the global variable in this file. You don't need to wrap the find method in a function and pass id as an argument